components/email-mgt/org.wso2.carbon.email.mgt/src/test/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManagerTest.java

```diff
@@ -0,0 +1,194 @@
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Class that contains the test cases for {@link InMemoryBasedTemplateManager}.
+ */
+@PrepareForTest({I18nMgtDataHolder.class, CarbonUtils.class})
+public class InMemoryBasedTemplateManagerTest extends PowerMockTestCase {
+
+    private final String baseDirectoryPath = Paths.get(System.getProperty("user.dir"),
+            "src", "test", "resources").toString();
+    private static final String tenantDomain = "carbon.super";
+    private static final String dummyDisplayName = "dummyDisplayName";
+    private static final String dummyAppId = "dummyAppId";
+    private static final String EN_US = "en_US";
+    @Mock
+    I18nMgtDataHolder i18nMgtDataHolder;
+    InMemoryBasedTemplateManager inMemoryBasedTemplateManager;
+    NotificationTemplate positiveNotificationTemplate;
+    NotificationTemplate negativeNotificationTemplate;
+
+    @BeforeMethod
+    public void setUp() {
+
+        initMocks(this);
+        mockStatic(I18nMgtDataHolder.class);
+        i18nMgtDataHolder = PowerMockito.mock(I18nMgtDataHolder.class);
+        when(I18nMgtDataHolder.getInstance()).thenReturn(i18nMgtDataHolder);
+
+        List<NotificationTemplate> defaultEmailTemplate = loadDefaultTemplatesFromFile(baseDirectoryPath,
+                NotificationChannels.EMAIL_CHANNEL.getChannelType());
+        when(i18nMgtDataHolder.getDefaultEmailTemplates()).thenReturn(defaultEmailTemplate);
+
+        inMemoryBasedTemplateManager = new InMemoryBasedTemplateManager();
+        initTestNotificationTemplates();
+    }
+
+    private void initTestNotificationTemplates() {
+
+        positiveNotificationTemplate = i18nMgtDataHolder.getDefaultEmailTemplates().get(0);
+        negativeNotificationTemplate = new NotificationTemplate();
+        negativeNotificationTemplate.setNotificationChannel(NotificationChannels.EMAIL_CHANNEL.getChannelType());
+        negativeNotificationTemplate.setType("dummyType");
+        negativeNotificationTemplate.setDisplayName("dummyDisplayName");
+        negativeNotificationTemplate.setLocale("en_US");
+    }
+
+    @Test
+    public void testIsNotificationTemplateTypeExists() throws Exception {
+
+        assertFalse(inMemoryBasedTemplateManager.isNotificationTemplateTypeExists(StringUtils.EMPTY,
+                NotificationChannels.EMAIL_CHANNEL.getChannelType(), tenantDomain));
+        assertFalse(inMemoryBasedTemplateManager.isNotificationTemplateTypeExists(
+                negativeNotificationTemplate.getDisplayName(),
+                NotificationChannels.EMAIL_CHANNEL.getChannelType(), tenantDomain));
+        assertTrue(inMemoryBasedTemplateManager.isNotificationTemplateTypeExists(
+                positiveNotificationTemplate.getDisplayName(),
+                NotificationChannels.EMAIL_CHANNEL.getChannelType(), tenantDomain));
+    }
+
+    @Test
+    public void testListNotificationTemplateTypes() throws Exception {
```

Comment:
Add test/assert to verify the there are no duplicates records

Reply:
ack

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/commits/8628c42d5f74b1e58a734be7fc641c3f846af46d